src/dlite-collection.c

```diff
@@ -645,7 +669,31 @@ int dlite_collection_count(DLiteCollection *coll)
   int count=0;
   TripleState state;
   triplestore_init_state(coll->rstore, &state);
-  while (triplestore_find(&state, NULL, "_is-a", "Instance")) count++;
+  while (triplestore_find(&state, NULL, "_is-a", "Instance", NULL)) count++;
   triplestore_deinit_state(&state);
   return count;
 }
+
+
+/*
+  Return pointer to the value for a pair of two criteria.
+
+  Useful if one knows that there may only be one value.  The returned
+  value is held by the collection and should be copied by the user
+  since it may be overwritten by later calls to the collection.
+
+  Parameters:
+      s, p, o: Criteria to match. Two of these must be non-NULL.
+      d: If not NULL, the required datatype of literal objects.
+      fallback: Value to return if no matches are found.
+      any: If non-zero, return first matching value.
+
+  Returns a pointer to the value of the `s`, `p` or `o` that is NULL.
+  On error NULL is returned.
+ */
```

Comment:
I think this is a nice description. However, I have read several variations on the same theme further up. Does this mean that we are documenting the same thing several places? This is a bit confusing as well as increases the risk of updating documentation of the same thing wrongly (not in all pleaces) if needed later.

Reply:
Yes, we are duplicating documentation. In C we have the same docstring both in the source (.c) and header (.h) file. Function expose in Python will also have a Python documentation. 

It is possible to reduce this duplication. Currently we generate the C reference documentation from the header files, so we need documentation there. At the same time, it is very useful to have the function documentation in the source files, so we also need it there.

A possibility could be to We could remove all function documentation from the header files and update the documentation in the source files such that it can be parsed by doxygen. That would increase maintainability and could be added as an issue.

It would in principle also be possible to extract function documentation from the doxygen-generated xml file and auto-document Python functions from it. But firstly, it would increase the complexity and make dlite even more difficult understand and secondly we want to rephrase the documentation for Python as you commented about above. So this is not something I would suggest.